### ISSUE TITLE
Document EventId values in traces

### ISSUE THREAD

**Issue by missymessa** [dotnet/arcade#11366]
There are 32 distinct values for EventId when running this query in staging, and 38 distinct values in prod:

```
traces
| distinct toint(customDimensions.EventId)
```

None of these values are documented anywhere, which means someone investigating an event would need to rely on tribal knowledge to know which EventId to search for.

**Comment by garath** [comment]
What's the ask here?

**Comment by missymessa** [comment]
> What's the ask here?

Edited initial comment

**Comment by garath** [comment]
Which database are you looking?

**Comment by missymessa** [comment]
> Which database are you looking?

AppInsights, dotnet-eng and dotnet-eng-int

**Comment by garath** [comment]
This is interesting. I have not looked closely at event IDs before. 

I had a look at dotnet-eng and I don't think any of these are coming from our code. 

[Go to Log Analytics and run query](https://ms.portal.azure.com#@72f988bf-86f1-41af-91ab-2d7cd011db47/blade/Microsoft_OperationsManagementSuite_Workspace/Logs.ReactView/resourceId/%2Fsubscriptions%2F68672ab8-de0c-40f1-8d1b-ffb20bd62c0f%2FresourceGroups%2Fdotnet-eng-cluster%2Fproviders%2Fmicrosoft.insights%2Fcomponents%2Fdotnet-eng/source/LogsBlade.AnalyticsShareLinkToQuery/q/H4sIAAAAAAAAA4WQwW6DMAyG730KH0FCu%252BzMLnSTdlmfwQWLRVpsZJu2VH34hQ0GVNqWQ6TE3xfl%252F12xJtvd4PxOSuAhkjnGDp4AW8kemzzN6OLEzQ7Sej4R%252B2sDJbgE9qzuzSXuk8YWhO1hAvJiwd8w0pdgroHbX5yRmqwKnVrR4V9xDU7uQUMbGD9eRCP6n%252FYWzX9KCMbiFDsfsjlNmlkfI2q4ElTSs5f1uGc5HIe5lGKJW2wyFHe%252FSq%252BJNqQrF9DqrfR9sxSYjp8EWC49rQEAAA%253D%253D)
```kql
traces
| where timestamp > ago(3d)
| extend
    EventId = toint(customDimensions.EventId),
    EventName = tostring(customDimensions.EventName),
    CategoryName = tostring(customDimensions.CategoryName),
    OriginalFormat = tostring(customDimensions.OriginalFormat)
| where isnotempty(EventId)
| summarize Count=count() by EventId, EventName, CategoryName, OriginalFormat
| order by EventId asc, CategoryName asc, EventName asc
```

**Comment by missymessa** [comment]
There is at least one that will be coming from our code whenever it shows up: https://dnceng.visualstudio.com/internal/_git/dotnet-helix-service/commit/bad16983b05809d2614ca187f54ba4fdc4391f2c?refName=refs%2Fheads%2Fmain

**Comment by garath** [comment]
The entries in my previous query without a "CategoryName" all appear to be from Service Fabric (?). (The `any(*)` here isn't conclusive but it's a start.)

[Go to Log Analytics and run query](https://ms.portal.azure.com#@72f988bf-86f1-41af-91ab-2d7cd011db47/blade/Microsoft_OperationsManagementSuite_Workspace/Logs.ReactView/resourceId/%2Fsubscriptions%2F68672ab8-de0c-40f1-8d1b-ffb20bd62c0f%2FresourceGroups%2Fdotnet-eng-cluster%2Fproviders%2Fmicrosoft.insights%2Fcomponents%2Fdotnet-eng/source/LogsBlade.AnalyticsShareLinkToQuery/q/H4sIAAAAAAAAA%252B2STW7CQAxG9zmFlxOEumGJYNNSqRt6BpNYwVLGQ8dO2yAOzwQSftQCF8Db772x%252FGlqMvhqSI2D4KqmxTeJfZQKM7CIBWm2g581RQJjnzD0G5gDVsFNyjxl9GskZQZperUzA4u5olEL%252Fi1poul1femBfHzGl%252BjpIKhFluqG01G99YpGVYjtQ%252FES7N3PyBUL1u8herS79jWan0pglWDkN9a64ZqL7BhcbU5pyaldKWwoaJo9m%252F3b7HAjC7j%252FPmRHauM9Rt4SoLRulMOqHbw9r1jsy8gCAAA%253D)
```kql
let questionableEventIds = traces
| where timestamp > ago(3d)
| extend
    EventId = toint(customDimensions.EventId),
    EventName = tostring(customDimensions.EventName),
    CategoryName = tostring(customDimensions.CategoryName),
    OriginalFormat = tostring(customDimensions.OriginalFormat)
| where isnotempty(EventId)
| where isempty(CategoryName)
| distinct EventId;
traces
| where timestamp > ago(3d)
| extend
    EventId = toint(customDimensions.EventId),
    EventName = tostring(customDimensions.EventName),
    CategoryName = tostring(customDimensions.CategoryName),
    OriginalFormat = tostring(customDimensions.OriginalFormat)
| where EventId in (questionableEventIds)
| summarize any(*) by EventId
```

**Comment by garath** [comment]
Given that Event IDs are new, it seems to me that this should go to the dev WF epic. Agree?

**Comment by missymessa** [comment]
I don't agree. Just because one was added as a part of a investigation for MessageAbandoned alert doesn't mean that the DevWF v-team should be responsible for tracking down all existing EventIds.

**Comment by garath** [comment]
Ah, yes, I also meant that not-our-code should be excluded. Are you asking that _all_ event IDs be documented, even if they don't originate from our code?

**Comment by missymessa** [comment]
All EventIds should be documented. I'm not sure how they get into our telemetry if they don't originate from our code.

**Comment by garath** [comment]
App insights collects data from several sources. And some framework services and libraries (like HttpClientFactory) will grab loggers from Dependency Injection. 

I'm ambivalent about trying to document too much on things we don't own, especially technical details.

But we should definitely document our own event codes. Logging in helix is very... organic.

**Comment by ChadNedzlek** [comment]
They are documented in that there is a giant enum full of all the event ids we generate (so that they don't overlap).  I don't want to try to reproduce that list in some MD file that never gets updated, because it'll just be out-of-date all the time, and wrong documentation is worse than no documentation, in my opinion. I think the "normal" course for this will be that you'll be browsing around AppInsights, looking at traces, and notice one of interest that you want to scan for in more than one operation... the eventId will be right there, and you'll just copy paste it.  We can have a document that's basically just a link to the source file, but I'm not sure how helpful that is.

**Comment by missymessa** [comment]
I think if we reference where to find the code(s) that define the EventId, I think that should work. Basically, I don't want these to get lost into the Tribal Knowledge ether, so that only a couple people know how to find out which EventId to reference.

**Comment by michellemcdaniel** [comment]
Ask is now to make a doc/wiki that links to the code where these IDs are written down.
